proto/spacemesh/v1/smesher_types.proto

```diff
@@ -74,8 +77,8 @@ message PostData {
     string path = 1; // User provided path to create the post data files at
     uint64 data_size = 2; // Requested post data size
     bool   append = 3; // Append to existing files if they exist. Otherwise overwrite.
-    bool   throttle = 4; // Throttle down setup phase computations while user is interactive on ///system
-    ComputeEngineFlags compute_engine_flags = 5; // One or more available compute engines to be used for gpu post
+    bool   throttle = 4; // Throttle down setup phase computations while user is interactive on system
+    uint32 provider_id = 5; // A PostProvider id
```

Comment:
Why return just an ID here and not a `PostComputeProvider`?

Reply:
I was thinking about this, it is an option but I didn't add it because the user can easily call PostComputeProviders() and match this id with the response in the case he wants to know what provided was used.

Reply:
And in addition - this way the full node only needs to persist this id and not the other PostComputeProvider fields.

Reply:
right but what if the list changes and the id no longer maps to the same thing?

Reply:
it will happen only when you add/remove a gpu and in this case clients can call the method to get the gpus to see what's the current status of the system so I don't think is is a major issue - a bit of an edge case.